src/c_api.cpp

```diff
@@ -1886,6 +1936,51 @@ int LGBM_BoosterPredictForCSRSingleRow(BoosterHandle handle,
   API_END();
 }
 
+int LGBM_BoosterPredictForCSRSingleRowFastInit(BoosterHandle handle,
+                                               const int data_type,
+                                               const int64_t num_col,
+                                               const char* parameter,
+                                               FastConfigHandle *out_fastConfig) {
+  API_BEGIN();
+  if (num_col <= 0) {
+    Log::Fatal("The number of columns should be greater than zero.");
+  } else if (num_col >= INT32_MAX) {
+    Log::Fatal("The number of columns should be smaller than INT32_MAX.");
+  }
```

Comment:
I am a little confused about `ncol.int64`. It seems both CSR and Mat use the 32-bit ncol, why int64 is needed?

Reply:
Great catch @guolinke , in my questions before:

> The sparse predictors like `LGBM_BoosterPredictForCSR` take num_col as int64_t, but `LGBM_BoosterPredictForMatSingleRow` takes int32_t.

I hadn't noticed that pre-condition in the function receiving int64_t and thought both types had to be supported.

Just removed that and the profiling/ folder.